### PR TITLE
Feature/second autolocking trng55

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description

Features:
- Reworked how the lock system works, added a new lock strategy which is based upon LockableProperty to lock SceneObject's properties according to usage in steps. TRNG-55
- 

Improvements:
- Improved performance in Chapter selection UI
- StepInspector UI now splits behaviors and conditions into different tabs.
- 

Chores:
- Removed Jenkins files from repository, as switching to Azure is complete.

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How Has This Been Tested?
- Added unit tests for the locking feature
- 

**Test Configuration**:

### Checklist

- [ ] My code follows the [Coding Conventions](#coding-conventions)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Editor/Configuration/RuntimeConfiguratorEditor.cs [hunk 1]
```diff
@@ -70,6 +68,8 @@ public override void OnInspectorGUI()
             // Courses can change without recompile so we have to check for them.
             UpdateAvailableCourses();
 
+            configurator.LockSceneObjectsOnStart = EditorGUILayout.Toggle("Lock all SceneObject on Start", configurator.LockSceneObjectsOnStart);
```
Comment (SimonTheSourcerer) [inline Editor/Configuration/RuntimeConfiguratorEditor.cs:71]: does not work, should not be in there.

File: Runtime/Conditions/Condition.cs [hunk 2]
```diff
@@ -31,5 +32,10 @@ protected Condition()
                 return Data;
             }
         }
+
+        public virtual IEnumerable<LockablePropertyData> GetLockableProperties()
```
Comment (SimonTheSourcerer) [inline Runtime/Conditions/Condition.cs:36]: docu

File: Runtime/Configuration/RuntimeConfigurator.cs [hunk 3]
```diff
@@ -38,6 +37,9 @@ public sealed class RuntimeConfigurator : MonoBehaviour
 
         private IRuntimeConfiguration runtimeConfiguration;
 
+        [SerializeField]
```
Comment (SimonTheSourcerer) [inline Runtime/Configuration/RuntimeConfigurator.cs:40]: Not working as expected, can be removed.

File: Runtime/CourseRunner.cs [hunk 4]
```diff
@@ -62,6 +64,14 @@ public void Execute()
             {
                 RuntimeConfigurator.ModeChanged += HandleModeChanged;
 
+                if (RuntimeConfigurator.Instance.LockSceneObjectsOnStart)
```
Comment (SimonTheSourcerer) [inline Runtime/CourseRunner.cs:67]: Not working as expected, can be removed.

File: Runtime/Properties/ILockablePropertiesProvider.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+using Innoactive.Creator.Core.RestrictiveEnvironment;
+
+namespace Innoactive.Creator.Core
+{
+    public interface ILockablePropertiesProvider
```
Comment (SimonTheSourcerer) [inline Runtime/Properties/ILockablePropertiesProvider.cs:6]: docu

File: Runtime/Properties/LockableProperty.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
+﻿﻿using System;
+ using Innoactive.Creator.Core.SceneObjects;
+ using UnityEngine;
+
+namespace Innoactive.Creator.Core.Properties
+{
+    public abstract class LockableProperty : TrainingSceneObjectProperty, ILockable
```
Comment (SimonTheSourcerer) [inline Runtime/Properties/LockableProperty.cs:7]: docu

File: Runtime/Properties/LockablePropertyData.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
+﻿using Innoactive.Creator.Core.Properties;
+
+namespace Innoactive.Creator.Core.RestrictiveEnvironment
+{
+    public class LockablePropertyData
```
Comment (SimonTheSourcerer) [inline Runtime/Properties/LockablePropertyData.cs:5]: docu

File: Runtime/Properties/LockablePropertyReference.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Runtime.Serialization;
+using Innoactive.Creator.Core.Properties;
+using Innoactive.Creator.Core.SceneObjects;
+
+namespace Innoactive.Creator.Core.Behaviors
+{
+    [DataContract(IsReference = true)]
+    public class LockablePropertyReference
```
Comment (SimonTheSourcerer) [inline Runtime/Properties/LockablePropertyReference.cs:9]: docu

File: Runtime/RestrictiveEnvironment/PropertyReflectionHelper.cs [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+using Innoactive.Creator.Core.Configuration;
+using Innoactive.Creator.Core.Properties;
+using Innoactive.Creator.Core.RestrictiveEnvironment;
+using Innoactive.Creator.Core.SceneObjects;
+using Innoactive.Creator.Core.Utils;
+using UnityEngine;
+
+namespace Innoactive.Creator.Core
+{
+    public static class PropertyReflectionHelper
```
Comment (SimonTheSourcerer) [inline Runtime/RestrictiveEnvironment/PropertyReflectionHelper.cs:15]: should not be public, internal is enough

File: Runtime/Transition.cs [hunk 10]
```diff
@@ -139,5 +140,18 @@ public Transition()
                 };
             }
         }
+
+        public IEnumerable<LockablePropertyData> GetLockableProperties()
```
Comment (SimonTheSourcerer) [inline Runtime/Transition.cs:144]: docu

File: Editor/EditorReflectionUtils.cs [hunk 11]
```diff
@@ -9,7 +9,7 @@
 using UnityEditor.Callbacks;
 
 namespace Innoactive.CreatorEditor
-{
+   {
```
Comment (Maxsiom) [inline Editor/EditorReflectionUtils.cs]:
```suggestion
{
```

File: Runtime/Configuration/IRuntimeConfiguration.cs [hunk 12]
```diff
@@ -8,6 +9,7 @@ namespace Innoactive.Creator.Core.Configuration
     /// <summary>
     /// An interface for training runtime configurations. Implement it to create your own.
     /// </summary>
+    [Obsolete("To be more flexible with the Creator development we are switched to an abstract class as configuration base, consider using BaseRuntimeConfiguration.")]
```
Comment (SimonTheSourcerer) [inline Runtime/Configuration/IRuntimeConfiguration.cs:12]:
```suggestion
    [Obsolete("To be more flexible with the Creator development we switched to an abstract class as configuration base, consider using BaseRuntimeConfiguration.")]
```

File: Runtime/Configuration/RuntimeConfigWrapper.cs [hunk 13]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Innoactive.Creator.Core.SceneObjects;
+using UnityEngine;
+
+namespace Innoactive.Creator.Core.Configuration
+{
+    /// <summary>
+    /// This wrapper for <see cref="IRuntimeConfiguration"/> configurations ensures
+    /// that old interface based configurations can still be used.
```
Comment (SimonTheSourcerer) [inline Runtime/Configuration/RuntimeConfigWrapper.cs:9]:
```suggestion
    /// This is wrapper used for <see cref="IRuntimeConfiguration"/> configurations, which
    /// ensures that the old interface based configurations can still be used.
```

File: Runtime/Properties/PropertyReflectionHelper.cs [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+using Innoactive.Creator.Core.SceneObjects;
+using Innoactive.Creator.Core.Utils;
+using UnityEngine;
+
+namespace Innoactive.Creator.Core
+{
+    public static class PropertyReflectionHelper
+    {
+
+        public static List<LockablePropertyData> ExtractLockablesFromStep(IStep step)
```
Comment (SimonTheSourcerer) [inline Runtime/Properties/PropertyReflectionHelper.cs:18]: docu

**Pull request closed without merging**